src/Workspaces/Core/Portable/Editing/SyntaxEditor.cs

```diff
@@ -245,30 +236,10 @@ public void InsertAfter(SyntaxNode node, SyntaxNode newNode)
         private void CheckNodeInOriginalTreeOrTracked(SyntaxNode node)
         {
             if (node == null)
```

Comment:
>if (node == null) [](http://example.com/codeflow?start=12&length=17)

private method, is null check needed?

Reply:
probably not.  will check the NRT status of things here.

Reply:
I'm going to keep the null check.  This is a public API and someone can pass us null.  Doing the up front check is valuable to catch errors immediately (instead of in a deferred fashion when GetChangedRoot is called).

Reply:
I would think it would be better to guard directly in the public API methods, rather than depending on a private method to check for them. That way, all private methods can safely take advantage of the nullability specified.

Reply:
I would generally agree with the code pattern suggested by Todd, but we have a precedence in Roslyn at a bunch of places where we have many public API overloads with similar parameter(s) and we have a common helper invoked upfront by all the APIs that checks null-ness and throws the exception.

Reply:
I'm ok leaving as is.  All these functions have to make the same check.  So lifting it out is just more complexity.  It also hasn't been an issue so far, so i'd prefer to leave as is.  Many methods everywhere (in roslyn, bcl, etc.) defer to a helper method to do validation.  I see no problem with that :)